static-instrumenter/maven-plugin/src/main/java/io/opentelemetry/contrib/staticinstrumenter/plugin/maven/PackagingSupportFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.plugin.maven;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public final class PackagingSupportFactory {
```

Comment:
What's the reason for adding the spring boot/war handling logic here? Can't it be added to the main static instrumenter tool? If we add it to the maven plugin, we'll have to duplicate it in the gradle plugin too. And the standalone static instrumenter will be sort of incomplete, should anyone want to use it by itself (or implement their own tool, e.g. in SBT or Leiningen)

Reply:
The idea is that once we start implementing something besides M3 plugin, we should factor out the file handling logic into a plugin-core module or so, as most ops will be quite similar. Then we can decide on the final logic split between the instrumenter and plugin (cores).

Reply:
I'd argue that the static agent is that "something" -- it should be possible to use it without any build system plugin, e.g. in a makefile or something (I know, kind of an extreme example, but hey, you can use `javac` without maven or gradle). It is my opinion that the "standalone command line tool" should be truly standalone, and various build plugins should be just thin wrappers over that.
Still, I agree with you that it can be factored out separately in another PR, once #319 is merged -- let's not make this one any larger than it already is.

Reply:
Yep, let's not complicate this further ;)